### PR TITLE
Bug 1211622 - Introduce a reverse proxy in l10n.js to l20n.js

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1211622

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=af75f1d57dbef87de0091eb1232a025977678f7f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=42e1fa08ef0910b84990306045ae469c7d6f9241

File: shared/js/l10n.js [hunk 1]
```diff
@@ -2138,4 +2138,29 @@
...
+    }
+  };
+
+  navigator.mozL10n.once(() => {
+    window.addEventListener('localized', () => {
+      document.dispatchEvent(new CustomEvent('DOMRetranslated', {
+        bubbles: false,
+        cancelable: false,
+        detail: {
+          languages: navigator.mozL10n.languages
```
Comment (stasm) [inline shared/js/l10n.js]: `navigator.mozL10n.ctx.supportedLocales`?

File: shared/js/l10n.js [hunk 2]
```diff
@@ -2138,4 +2138,29 @@
     whenInteractive(init.bind(navigator.mozL10n, forcePretranslate));
   }
 
+  document.l10n = {
+    setAttributes: navigator.mozL10n.setAttributes,
+    getAttributes: navigator.mozL10n.getAttributes,
+    formatValue: navigator.mozL10n.formatValue.bind(navigator.mozL10n),
+    translateFragment: (...args) =>
+      Promise.resolve(navigator.mozL10n.translateFragment(...args)),
+    ready: () => {
```
Comment (stasm) [inline shared/js/l10n.js]: Nit: no need for curly braces.

File: shared/js/l10n.js [hunk 3]
```diff
@@ -2138,4 +2138,29 @@
     whenInteractive(init.bind(navigator.mozL10n, forcePretranslate));
   }
 
+  document.l10n = {
+    setAttributes: navigator.mozL10n.setAttributes,
+    getAttributes: navigator.mozL10n.getAttributes,
+    formatValue: navigator.mozL10n.formatValue.bind(navigator.mozL10n),
```
Comment (stasm) [inline shared/js/l10n.js]: Please use an arrow function here.

File: shared/js/l10n.js [hunk 4]
```diff
@@ -2138,4 +2138,29 @@
...
+      Promise.resolve(navigator.mozL10n.translateFragment(...args)),
+    ready: () => {
+      return new Promise(function(resolve, reject) {
+        navigator.mozL10n.once(resolve);
+      });
+    }
+  };
+
+  navigator.mozL10n.once(() => {
+    window.addEventListener('localized', () => {
```
Comment (stasm) [inline shared/js/l10n.js]: Nit: no need for curly braces.

File: shared/js/l10n.js [hunk 5]
```diff
@@ -2138,4 +2138,29 @@
...
 
+  document.l10n = {
+    setAttributes: navigator.mozL10n.setAttributes,
+    getAttributes: navigator.mozL10n.getAttributes,
+    formatValue: navigator.mozL10n.formatValue.bind(navigator.mozL10n),
+    translateFragment: (...args) =>
+      Promise.resolve(navigator.mozL10n.translateFragment(...args)),
+    ready: () => {
+      return new Promise(function(resolve, reject) {
+        navigator.mozL10n.once(resolve);
```
Comment (stasm) [inline shared/js/l10n.js]: `mozL10n.once` means: context is ready to use.  `l10n.ready` means: the DOM has been localized.  Should we take this difference into account here?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=99052523b8080fac85872020ed7df7d212a7659d

File: shared/js/l10n.js (hunk 5 above)
Comment (zbraniecki) [inline shared/js/l10n.js]: nah

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3394f6fcb2b3d57386f519f9c59c6d65281e9c85

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f8508b0ab7ec87ebaa9fc04590a2c642cd4ebe27

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=08db9f57b11792715571645f229a04e2d41b4d08

**Pull request merged**